runtime/runtime/src/lib.rs

```diff
@@ -2489,6 +2544,81 @@ mod tests {
             .expect("Compiled contract should be cached")
             .expect("Compilation result should be non-empty");
     }
+
+    #[cfg(feature = "protocol_feature_compute_costs")]
+    #[test]
+    fn test_compute_usage_limit() {
+        let initial_balance = to_yocto(1_000_000);
+        let initial_locked = to_yocto(500_000);
+        let (runtime, tries, root, mut apply_state, signer, epoch_info_provider) =
+            setup_runtime(initial_balance, initial_locked, 1);
+
+        let mut free_config = RuntimeConfig::free();
+        let sha256_cost =
+            ParameterCost { gas: Gas::from(1u64), compute: Compute::from(10_000_000_000_000u64) };
+        free_config.wasm_config.ext_costs.costs[ExtCosts::sha256_base] = sha256_cost.clone();
+        apply_state.config = Arc::new(free_config);
+        // This allows us to execute 1 receipt with a function call per apply.
+        apply_state.gas_limit = Some(sha256_cost.compute);
+
+        let mut receipts = create_receipts_with_actions(
+            alice_account(),
+            signer.clone(),
+            vec![Action::DeployContract(DeployContractAction {
+                code: near_test_contracts::rs_contract().to_vec(),
+            })],
+        );
+
+        for _ in 0..2 {
+            receipts.extend(create_receipts_with_actions(
+                alice_account(),
+                signer.clone(),
+                vec![Action::FunctionCall(FunctionCallAction {
+                    method_name: "ext_sha256".to_string(),
+                    args: b"world".to_vec(),
+                    gas: 1,
+                    deposit: 0,
+                })],
+            ));
+        }
+
+        let apply_result = runtime
+            .apply(
+                tries.get_trie_for_shard(ShardUId::single_shard(), root),
+                &None,
+                &apply_state,
+                &receipts,
+                &[],
+                &epoch_info_provider,
+                Default::default(),
+            )
+            .unwrap();
+        let mut store_update = tries.store_update();
+        let root = tries.apply_all(
+            &apply_result.trie_changes,
+            ShardUId::single_shard(),
+            &mut store_update,
+        );
+        store_update.commit().unwrap();
+
+        assert_eq!(apply_result.outcomes.len(), 2);
+        assert_eq!(apply_result.outcomes[1].outcome.compute_usage.unwrap(), sha256_cost.compute);
+
+        let apply_result = runtime
+            .apply(
+                tries.get_trie_for_shard(ShardUId::single_shard(), root),
+                &None,
+                &apply_state,
+                &[],
+                &[],
+                &epoch_info_provider,
+                Default::default(),
+            )
+            .unwrap();
+
+        assert_eq!(apply_result.outcomes.len(), 1);
+        assert_eq!(apply_result.outcomes[0].outcome.compute_usage.unwrap(), sha256_cost.compute);
```

Comment:
please also check that execution was successful

Reply:
Done.